src/main/java/net/snowflake/client/core/SnowflakeJdbcInternalApi.java

```diff
@@ -0,0 +1,19 @@
+package net.snowflake.client.core;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Elements marked with this annotation should be considered as internal API even if they are public
+ */
+@Target({
+  ElementType.PACKAGE,
+  ElementType.TYPE,
+  ElementType.FIELD,
+  ElementType.CONSTRUCTOR,
+  ElementType.METHOD
+})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface SnowflakeJdbcInternalApi {}
```

Comment:
In my opinion it will be better to create such interface in separated commit to stress the importance of new internal api marking.

Reply:
I will extract it to separate PR

Reply:
Extracted to https://github.com/snowflakedb/snowflake-jdbc/pull/1629